ui/fisher/seabase.js

```diff
@@ -0,0 +1,283 @@
+'use strict';
+
+class SeaBase {
+    constructor() {
+        this._dbName = 'seabase';
+        this._dbVersion = 1;
+        this._storeName = 'catches';
+        this.db = null;
+    }
+
+    getConnection() {
+        let _this = this;
+
+        return new Promise(function(resolve, reject) {
+            let req = window.indexedDB.open(_this._dbName, _this._dbVersion);
+
+            req.onsuccess = function(event) {
+                resolve(this.result);
+            }
+
+            req.onerror = function(event) {
+                reject(this.error);
+            }
+
+            req.onupgradeneeded = function(event) {
+                let db = event.target.result;
+                let tx = event.target.transaction;
+                let objectStore;
+
+                if (!db.objectStoreNames.contains(_this._storeName)) {
+                    objectStore = db.createObjectStore(_this._storeName, {autoIncrement: true});
+                } else {
+                    objectStore = tx.objectStore(_this._storeName);
+                }
+
+                if (!objectStore.indexNames.contains('fish')) {
+                    objectStore.createIndex('fish', 'fish', { unique: false });
+                }
+
+                if (!objectStore.indexNames.contains('fishbaitchum')) {
+                    objectStore.createIndex('fishbaitchum', ['fish', 'bait', 'chum'], { unique: false });
+                }
+
+                tx.oncomplete = function(event) {
+                    resolve(db);
+                }
+            };
+        });
+    }
+
+    getTransaction(db, mode) {
+        return db.transaction(this._storeName, mode);
+    }
+
+    getIQRThresholds(times) {
```

Comment:
I haven't ended up with any. The effort here is mostly because I'm nervous taking the absolute minimum and absolute maximum in a data set; for example, maybe there's a lag spike at some point that drags out the time window significantly, and then it's stuck that way forever. I google the issue, found the IQR method for doing so, and decided to use that. If it was on a larger scale, I'd probably just drop the extreme 5% on the ends, but doing that moment-to-moment seems like it might eventually take a while to process.